### DISCUSSION TITLE
Unable to use @DependencyEndpoint with a typealias

### DISCUSSION THREAD

**Discussion by JOyo246** [pointfreeco/swift-dependencies#179] [General]
### Description

Not sure if this is a dependencies bug, macros bug, or macros feature?

**works!**
```
extension MyFeature {
  @DependencyClient
  struct Client {
    var refreshRequest: @Sendable (MyFeaturesRequest) async throws -> MyFeaturesRequest.Content
  }
}
```

-------

**Doesnt work**
```
extension MyFeature {
  @DependencyClient
  struct Client {
    var refreshRequest: RequestOf<MyFeaturesRequest>
  }
}

typealias RequestOf<R: NetworkRequestProtocol> = @Sendable (R) async throws -> R.Content
```

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-dependencies/issues) or [discussion](https://github.com/pointfreeco/swift-dependencies/discussions).

### Expected behavior

I'd expect the DependencyEndpoint macro to work, when the typealias is representing a closure.

### Actual behavior

The DependencyEndpoint macro doesn't apply when the typealias is representing a closure.

**Comment by mbrandonw** [comment]
Hi @JOyo246, this is a limitation of macros. The `@DependencyClient` macro can only see the syntax it is directly attached to, so it has no idea what `RequestOf` is. So unfortunately you cannot use type aliases in this way.

Since this isn't an issue with the library I am going to convert it to a discussion.

**Reply by JOyo246** [reply]
Thanks for the speedy reply!
